### ISSUE TITLE
Syntax highlighting incorrect for methods with type hinting for complex return.

### ISSUE THREAD

**Issue by bedge** [MagicStack/MagicPython#162]
VS Code version: Version 1.28.0-insider (1.28.0-insider)
vsc python Extension version: 2018.8.0
vsc magicpython Extension version: 1.0.12
OS and version: MacOS High Sierra 10.13.6 (17G65)
Python version (& distribution if applicable, e.g. Anaconda): Virtualenv 3.6
Type of virtual environment used (N/A | venv | virtualenv | conda | ...): virtualenv

Expected behavior
get_args method comment, and string should be formatted as they are in get_args_no_hints

![screen shot 2018-10-02 at 4 27 12 pm](https://user-images.githubusercontent.com/499317/46382718-57163280-c661-11e8-87c8-b12906e41a41.png)

Similar after installing Chromodynamics theme:
![screen shot 2018-10-02 at 4 50 22 pm](https://user-images.githubusercontent.com/499317/46383075-567e9b80-c663-11e8-941c-f5f3213ded96.png)

Steps to reproduce:
Add type hints to method. The return value of anything other than a simple scalar causes this.

https://github.com/Microsoft/vscode-python/issues/2768

**Comment by elprans** [comment]
This is highlighted with MagicPython (correctly):

```python
def get_args(arglist: List[str]) -> Dict[str, str]:
    """
    comment
    """
    return args  
```

What does the bottom-left corner of the status bar say?

**Comment by bedge** [comment]
Nothing relevant. I have many extensions loaded, so there's a lot of status bits from many places.
I checked the extension logs and didn't see anything interesting, even when I added/removed the return type annotation and the highlighting toggled from bad/good.
Restarting with all extensions disabled the highlighting is correct. I guess I'll disable one at a time to see what's causing this. 
I'll report back.
